### PR TITLE
docs: RemoveEffect interactive example

### LINKED ISSUE #1957: Interactive docs: `RemoveEffect`
# Interactive doc for the `RemoveEffect`

The `RemoveEffect` should have a small example that show how the effect is applied and how it works. This could for example be just clicking somewhere in the game and then have a count down and after that count-down remove a geometrical shape.

https://docs.flame-engine.org/1.3.0/flame/effects.html#removeeffect

# What are interactive docs?

On our documentation site has the possibility to run inline interactive examples and we would like every visual feature to have a **small**, pretty and easily understandable example attached to it. Some current examples are the [router](https://docs.flame-engine.org/1.3.0/flame/router.html) and the [decorators](https://docs.flame-engine.org/1.3.0/flame/rendering/decorators.html).
Some inspiration can also be found on our [examples page](https://examples.flame-engine.org) where we have examples for most features.

# Good to know

Make sure that you comment on the issue that you will be working on this before starting to program.

To set up the Flame environment you need to follow the instructions in our [contributing.md](https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md#environment-setup) doc, if you have any problems setting up your environment (or other issues of course) feel free to ask for help in the Flame channel on our [Discord server](https://discord.com/invite/pxrBmy4).

Since this is a documentation change, you need to be able to build the docs. You can do that by running `melos sphinxdoc` after you have followed the instructions in the contributing.md doc and also [installed Sphinx on your system](https://www.sphinx-doc.org/en/master/usage/installation.html). After you have built the docs you can open the `doc/_build/html/index.html` file in your web browser to see the changes.

Make sure to include a video of the example when you upload the PR so that it will be easier for us to review it. (You can attach videos directly in the PR, so no need to upload them somewhere else)

**Happy hacking!**

### PR DESCRIPTION
**Pull request by idlehart**
# Description
Added interactive example for `RemoveEffect`.

Added a simple way to re-add the flower to the example by a simple tap like you can reset the other examples.

## Checklist

- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [x] I have followed the [Contributor Guide] when preparing my PR.
- [ ] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [ ] Yes, this PR is a breaking change.
- [x] No, this PR is not a breaking change.

## Related Issues

Closes #1957 

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org

### CODE REVIEW

File: doc/flame/examples/lib/remove_effect.dart [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                    EffectController(duration: 0.75),
+                  ),
+                ),
+            );
+          }
+        },
+      )
+        ..anchor = Anchor.topLeft
+        ..size = size,
+    );
```
Comment (spydon) [inline doc/flame/examples/lib/remove_effect.dart:40]: I would just use `TapDetector` on the game instead and use that to control the taps by overriding `onTapDown`, and I don't think the `OpacityEffect` is needed, better keep the code as short as possible for the example.

Comment (spydon) [reply doc/flame/examples/lib/remove_effect.dart:40]: Possibly having a `TextComponent` that shows how long it is left until it will be removed?

**Comment by idlehart** [comment]
Text not working at the moment, will look more tomorrow.

**Comment by idlehart** [comment]
Should work properly now :)

File: doc/flame/examples/lib/remove_effect.dart [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+class RemoveEffectGame extends FlameGame with TapDetector {
+  static const double delayTime = 3;
+  late EmberPlayer flower;
+  late TextComponent textComponent;
+  late RemoveEffect effect = RemoveEffect(delay: delayTime);
+
+  @override
+  Future<void> onLoad() async {
+    add(
+      flower = EmberPlayer(
```
[the remark this answers is not in the document]
Comment (idlehart) [reply doc/flame/examples/lib/remove_effect.dart]: Silly miss, fixed :)

File: doc/flame/examples/lib/remove_effect.dart [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import 'package:flame/components.dart';
+import 'package:flame/effects.dart';
+import 'package:flame/events.dart';
+import 'package:flame/game.dart';
+
+class RemoveEffectGame extends FlameGame with TapDetector {
+  static const double delayTime = 3;
+  late EmberPlayer ember;
+  late TextComponent textComponent;
+  late RemoveEffect effect = RemoveEffect(delay: delayTime);
```
Comment (spydon) [inline doc/flame/examples/lib/remove_effect.dart:11]:
```suggestion
  final RemoveEffect effect = RemoveEffect(delay: delayTime);
```

**Review by spydon — APPROVED** [review]

**Pull request merged**